pandas/core/window/rolling.py

```diff
@@ -247,8 +247,31 @@ def __repr__(self) -> str:
         return f"{self._window_type} [{attrs}]"
 
     def __iter__(self):
-        url = "https://github.com/pandas-dev/pandas/issues/11704"
-        raise NotImplementedError(f"See issue #11704 {url}")
+        window = self._get_window(win_type=None)
+
+        blocks, obj = self._create_blocks()
+        index = self._get_window_indexer(window=window)
+
+        # Choose the min between min_periods and window to determine the output size
+        if self.min_periods is None:
+            iter_threshold = window
+        else:
+            iter_threshold = min(window, self.min_periods)
+
+        start, end = index.get_window_bounds(
+            num_values=len(obj),
+            min_periods=self.min_periods,
+            center=self.center,
+            closed=self.closed,
+        )
+        # From get_window_bounds, those two should be equal in length of array
+        assert len(start) == len(end)
+
+        window_size = len(start)
+        for i in range(window_size):
```

Comment:
Nit: Could we just do `for s, e in zip(start, end):`

Reply:
nice one! changed